Core/contentblocker.js

```diff
@@ -18,13 +18,38 @@
 //
 
 (function() {
+    
+    function throttle (callback, limit) {
+        var waiting = false;                      // Initially, we're not waiting
+        return function () {                      // We return a throttled function
+            if (!waiting) {                       // If we're not waiting
+                callback.apply(this, arguments);  // Execute users function
+                waiting = true;                   // Prevent future invocations
+                setTimeout(function () {          // After a period of time
+                    waiting = false;              // And allow future invocations
+                }, limit);
+            }
```

Comment:
I wonder if following scenario is correct or there may be a bug:

1. Add message 'one.com' , Invoke `throttledSend()` -> `waiting` set to 'true', 'one.com' delivered to native layer, array cleared.
2. Add message 'two.com', Invoke `throttleSend()` while `waiting` is still set to true.

And I right that unless there is another call to `throttleSend()`, 'two.com' won't be ever delivered to native layer?

Reply:
Correct. Fixed this.